engine/src/main/java/org/terasology/rendering/primitives/BlockMeshGeneratorSingleShape.java

```diff
@@ -45,9 +41,63 @@ public BlockMeshGeneratorSingleShape(Block block) {
 
     @Override
     public void generateChunkMesh(ChunkView view, ChunkMesh chunkMesh, int x, int y, int z) {
-        Biome selfBiome = view.getBiome(x, y, z);
-        Block selfBlock = view.getBlock(x, y, z);
+        final Block selfBlock = view.getBlock(x, y, z);
+
+        // Gather adjacent blocks
+        final Map<Side, Block> adjacentBlocks = Maps.newEnumMap(Side.class);
+        for (Side side : Side.getAllSides()) {
+            Vector3i offset = side.getVector3i();
+            Block blockToCheck = view.getBlock(x + offset.x, y + offset.y, z + offset.z);
+            adjacentBlocks.put(side, blockToCheck);
+        }
+
+        for (final Side side : Side.getAllSides()) {
+            if (isSideVisibleForBlockTypes(adjacentBlocks.get(side), selfBlock, side)) {
+                final Biome selfBiome = view.getBiome(x, y, z);
```

Comment:
Yes, the biome doesn't change. I would consider the multiple fetching a "bug" from my refactoring which does more work than it needs to.
The intention of the move inside the if statement was to only load the biome if we need it for processing. So instead of loading it in the beginning of the method I would make a change to initialize the biome with null outside of the loop and then guard the `getBiome` call with a null check to only load it in the first iteration where it is needed.